src/server/req_runjob.c

```diff
@@ -748,19 +758,8 @@ req_runjob2(struct batch_request *preq, job *pjob)
 	/* Check if prov is required, if so, reply_ack and let prov finish */
 	/* else follow normal flow */
 	prov_rc = check_and_provision_job(preq, pjob, &need_prov);
-
-	/* In case of subjob, save it to the database now because
-	 * not saved to the database so far.
-	 */
-	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) {
-		if (job_save_db(pjob)) {
-			free_nodes(pjob);
-			req_reject(PBSE_SAVE_ERR, 0, preq);
-			return;
-		}
-	}
-
 	if (prov_rc) { /* problem with the request */
+		save_subjob(pjob);
```

Comment:
if you fail to save, you want to reject with the DB save error?

Reply:
this was intentional, we are in an error case anyways, I think it's better to return the original error than return the db save error, that's why i didn't handle it. Let me know what you think.

Reply:
Sounds fine